CRM/Member/Form/MembershipType.php

```diff
@@ -187,12 +187,11 @@ public function buildQuickForm() {
 
     $membershipRecords = FALSE;
     if ($this->_action & CRM_Core_Action::UPDATE) {
-      $membershipType = new CRM_Member_BAO_Membership();
-      $membershipType->membership_type_id = $this->_id;
-      if ($membershipType->find(TRUE)) {
-        $membershipRecords = TRUE;
-        $memberRel->freeze();
-      }
+        $result = civicrm_api3("Membership", "get", ["membership_type_id"=>$this->_id, "options"=>["limit"=>1]]);
+        $membershipRecords = $result["count"] > 0;
+        if ($membershipRecords) {
+            $memberRel->freeze();
```

Comment:
MembershipRecords still gets set, only a couple of lines earlier (L191). I added a commit to change the array style.